### ISSUE TITLE
build-native.sh fails when using uname to determine arch

### ISSUE THREAD

**Issue by buddyabaddon** [dotnet/corefx#28761]
In src/Native/build-native.sh, the following logic fails when the processor type info contains spaces.
```
CPUName=$(uname -p)
# Some Linux platforms report unknown for platform, but the arch for machine.
if [ $CPUName == "unknown" ]; then
    CPUName=$(uname -m)
fi

if [ $CPUName == "i686" ]; then
    __BuildArch=x86
fi
```
```uname -p``` does not return the 'platform' as the comment seems to imply but rather the 'processor' type. While it appears some distros do return the architecture, others provide the processor type information correctly if available.

Examples:
```
> uname -p
Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz
> uname -p
Intel(R) Core(TM)2 Duo CPU E6750 @ 2.66GHz
```
Note that the returned data can contain spaces. This causes the conditional checks above to fail with the following error as they don't quote the variable when referencing it.
```
source-build/src/corefx/src/Native/build-native.sh: line 178: [: too many arguments
source-build/src/corefx/src/Native/build-native.sh: line 182: [: too many arguments
```

**Comment by buddyabaddon** [comment]
It looks like $CPUName is later referenced to see if cross building is needed. The expectation is then that this contains the build architecture (i686, x86_64, etc). In that case, ```uname -p``` shouldn't be used at all here.

**Comment by wfurt** [comment]
I think CPUName=$(uname -m) should be correct. @buddyabaddon  to confirm. 
It returns amd64 on FreeBSD (same -p) and x86_64 on OSX10.12 (instead of i386)

It would be interesting to know what kernel version or distro does this.

**Comment by buddyabaddon** [comment]
@danmosemsft I did offer a fix: #28762

@wfurt It appears Ubuntu (verified in a docker container) returns the arch for both 'uname -p' and 'uname -m' which I'm guessing is why this code is originally like this.
I ran this on a Gentoo distro which does what I would expect to be the correct thing and return the processor type for 'uname -p' and the arch infor for 'uname -m'.
